### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by DomeSs97** [thejoabo/autofishbot#56]
[*] Starting...
[*] Loading "test.config" ...
[*] Connecting to gateway ...
[*] Starting heartbeat thread ...
[*] Loging in ...
[*] Session stablished !
{'code': 50035, 'errors': {'data': {'_errors': [{'code': 'COMPONENT_VALIDATION_FAILED', 'message': 'Component validation failed'}]}}, 'message': 'Invalid Form Body'}
{'code': 50035, 'errors': {'data': {'_errors': [{'code': 'COMPONENT_VALIDATION_FAILED', 'message': 'Component validation failed'}]}}, 'message': 'Invalid Form Body'}
[E] "Receiver" thread exited.

**Comment by thejoabo** [comment]
Hello @DomeSs97 , can you provide more details about this error ? At which frequency it's happening ? Did it happen after you pressed a keybind to send a command, or the bot just crashed ? I still have to implement the debug function for requests, with these logs alone I can't figure out what's wrong. 

If the problem is too frequent, rollback to the stable version [here](https://github.com/thejoabo/autofishbot/blob/main/assets/stable-legacy/stable-1.2.1.1.zip).

**Comment by DomeSs97** [comment]
Hi, the bug comes out of nowhere, after 1hour of playing, after 2 hours, after 6 hours. variously when the bot is running in the background or open in a window.

**Comment by paulorica54** [comment]
Same error that happens to me

**Comment by thejoabo** [comment]
@DomeSs97 , @paulorica54
So, I released a quick patch to log those errors in more detail, please update to the latest version, enable debug mode in your config and try to replicate the error. Once it happens, find the "debug.log" file under '{autofishbot's directory}/app/debug.log' and search for the "ax0" and "ax1" identifiers (the message should also appear in your console). After that paste the log here so I can understand why the request failed.

**Note: this log will contain some sensitive information (such as your discord channel id and guild id), please, remove all private information.**

**Comment by DomeSs97** [comment]
[17:00 - 11/22/22] ax0 | [Traceback] [] | [Event] Code: 400
Response: {'code': 50035, 'errors': {'data': {'_errors': [{'code': 'COMPONENT_VALIDATION_FAILED', 'message': 'Component validation failed'}]}}, 'message': 'Invalid Form Body'}
Request: {'type': 3, 'nonce': 1044643463057902522, 'guild_id': '779792159976456263', 'channel_id': '779806780686860298', 'message_flags': 0, 'message_id': '1044643777970712587', 'application_id': '574652751745777665', 'data': {'component_type': 2, 'custom_id': 'fish 973271822'}, 'session_id': 'hCVww8pfL78y0CyQTO6dOeDQFVWf6EpK'}
Headers: {'Date': 'Tue, 22 Nov 2022 16:01:32 GMT', 'Content-Type': 'application/json', 'Content-Length': '165', 'Connection': 'keep-alive', 'set-cookie': '__dcfduid=eed409ba6a7e11edbed74ad907dcb126; Expires=Sun, 21-Nov-2027 16:01:32 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/, __sdcfduid=eed409ba6a7e11edbed74ad907dcb12687d7902c369fe2b391fa760c298aa23a3150f5cd23aca8bb74d06dfe62b2f425; Expires=Sun, 21-Nov-2027 16:01:32 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/, __cfruid=cad5435477afd3f14a7d43e290fe9644585647d7-1669132892; path=/; domain=.discord.com; HttpOnly; Secure; SameSite=None', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'x-envoy-upstream-service-time': '73', 'Via': '1.1 google', 'Alt-Svc': 'h3=":443"; ma=86400, h3-29=":443"; ma=86400', 'CF-Cache-Status': 'DYNAMIC', 'Report-To': '{"endpoints":[{"url":"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=8nb044dX%2FnpbVGUUrDWn%2FdCqWB4Oxcypr8ITGD3frS4EbZ4c5krqBqoFd3etgG8yOcPv65x2TvQ94XnzRNtSLYFizwNOn7N6f95hk%2Bvs8nD%2FdPecgjJIpG%2F4x4Mp"}],"group":"cf-nel","max_age":604800}', 'NEL': '{"success_fraction":0,"report_to":"cf-nel","max_age":604800}', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '76e2e2606bd327b8-PRG'} | [Stack] [FrameInfo(frame=<frame at 0x000001EB83A1BB40, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\utils.py', line 50, code log>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\utils.py', lineno=50, function='log', code_context=["            log_data = f'\\n[{convert_time(time())}] {id} | [Traceback] {trace()} | [Event] {event} | [Stack] {stack()}\\n'\n"], index=0, positions=Positions(lineno=50, end_lineno=50, col_offset=110, end_col_offset=117)), FrameInfo(frame=<frame at 0x000001EB8394BCC0, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\api_wrapper.py', line 397, code request>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\api_wrapper.py', lineno=397, function='request', code_context=["                        debugger.log(err_message, 'ax0')\n"], index=0, positions=Positions(lineno=397, end_lineno=397, col_offset=24, end_col_offset=56)), FrameInfo(frame=<frame at 0x000001EB83A95300, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\autofishbot.py', line 288, code run>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\autofishbot.py', lineno=288, function='run', code_context=['                        if self.session.request(message_id=self.message.id, custom_id=self.message.play_id, category=BUTTON):\n'], index=0, positions=Positions(lineno=288, end_lineno=288, col_offset=27, end_col_offset=124)), FrameInfo(frame=<frame at 0x000001EB83AF2980, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 975, code run>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=975, function='run', code_context=['                self._target(*self._args, **self._kwargs)\n'], index=0, positions=Positions(lineno=975, end_lineno=975, col_offset=16, end_col_offset=57)), FrameInfo(frame=<frame at 0x000001EB83AF2380, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 1038, code _bootstrap_inner>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=1038, function='_bootstrap_inner', code_context=['                self.run()\n'], index=0, positions=Positions(lineno=1038, end_lineno=1038, col_offset=16, end_col_offset=26)), FrameInfo(frame=<frame at 0x000001EB83A89FE0, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 995, code _bootstrap>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=995, function='_bootstrap', code_context=['            self._bootstrap_inner()\n'], index=0, positions=Positions(lineno=995, end_lineno=995, col_offset=12, end_col_offset=35))]

[17:14 - 11/22/22] ax0 | [Traceback] [] | [Event] Code: 400
Response: {'code': 50035, 'errors': {'data': {'_errors': [{'code': 'COMPONENT_VALIDATION_FAILED', 'message': 'Component validation failed'}]}}, 'message': 'Invalid Form Body'}
Request: {'type': 3, 'nonce': 1044647193857756408, 'guild_id': '779792159976456263', 'channel_id': '779806780686860298', 'message_flags': 0, 'message_id': '1044645194420723745', 'application_id': '574652751745777665', 'data': {'component_type': 2, 'custom_id': 'fish 1960018393'}, 'session_id': 'hCVww8pfL78y0CyQTO6dOeDQFVWf6EpK'}
Headers: {'Date': 'Tue, 22 Nov 2022 16:16:21 GMT', 'Content-Type': 'application/json', 'Content-Length': '165', 'Connection': 'keep-alive', 'set-cookie': '__dcfduid=010ce1866a8111edbe58369ce8409e52; Expires=Sun, 21-Nov-2027 16:16:21 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/, __sdcfduid=010ce1866a8111edbe58369ce8409e52d07ffa127d27e91fba8d82fa65f41f27126b8af48c86ff631e167c8959dafae5; Expires=Sun, 21-Nov-2027 16:16:21 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/, __cfruid=ed1a0e1ff0492d4f3e241d59c375d544eab1aa91-1669133781; path=/; domain=.discord.com; HttpOnly; Secure; SameSite=None', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'x-envoy-upstream-service-time': '135', 'Via': '1.1 google', 'Alt-Svc': 'h3=":443"; ma=86400, h3-29=":443"; ma=86400', 'CF-Cache-Status': 'DYNAMIC', 'Report-To': '{"endpoints":[{"url":"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=IMIk6WuL4dzgMWqsE2Gfgo%2BQEYwK71pJIljr9ear7Wfw%2Bl1G3%2BmEio%2Bgp8UEqZWoE7aGpjjqH3mzKDuIPIKjnpkTsbEaaP4864j5BmW%2BUhB1hmJtJh6X6GBOnVxv"}],"group":"cf-nel","max_age":604800}', 'NEL': '{"success_fraction":0,"report_to":"cf-nel","max_age":604800}', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '76e2f817df36b35d-PRG'} | [Stack] [FrameInfo(frame=<frame at 0x000001EB83A1A840, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\utils.py', line 50, code log>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\utils.py', lineno=50, function='log', code_context=["            log_data = f'\\n[{convert_time(time())}] {id} | [Traceback] {trace()} | [Event] {event} | [Stack] {stack()}\\n'\n"], index=0, positions=Positions(lineno=50, end_lineno=50, col_offset=110, end_col_offset=117)), FrameInfo(frame=<frame at 0x000001EB8394BCC0, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\api_wrapper.py', line 397, code request>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\api_wrapper.py', lineno=397, function='request', code_context=["                        debugger.log(err_message, 'ax0')\n"], index=0, positions=Positions(lineno=397, end_lineno=397, col_offset=24, end_col_offset=56)), FrameInfo(frame=<frame at 0x000001EB83A95300, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\autofishbot.py', line 288, code run>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\autofishbot.py', lineno=288, function='run', code_context=['                        if self.session.request(message_id=self.message.id, custom_id=self.message.play_id, category=BUTTON):\n'], index=0, positions=Positions(lineno=288, end_lineno=288, col_offset=27, end_col_offset=124)), FrameInfo(frame=<frame at 0x000001EB83AF2980, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 975, code run>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=975, function='run', code_context=['                self._target(*self._args, **self._kwargs)\n'], index=0, positions=Positions(lineno=975, end_lineno=975, col_offset=16, end_col_offset=57)), FrameInfo(frame=<frame at 0x000001EB83AF2380, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 1038, code _bootstrap_inner>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=1038, function='_bootstrap_inner', code_context=['                self.run()\n'], index=0, positions=Positions(lineno=1038, end_lineno=1038, col_offset=16, end_col_offset=26)), FrameInfo(frame=<frame at 0x000001EB83A89FE0, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 995, code _bootstrap>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=995, function='_bootstrap', code_context=['            self._bootstrap_inner()\n'], index=0, positions=Positions(lineno=995, end_lineno=995, col_offset=12, end_col_offset=35))]

[17:55 - 11/22/22] ax0 | [Traceback] [] | [Event] Code: 400
Response: {'code': 50035, 'errors': {'data': {'_errors': [{'code': 'COMPONENT_VALIDATION_FAILED', 'message': 'Component validation failed'}]}}, 'message': 'Invalid Form Body'}
Request: {'type': 3, 'nonce': 1044657311085237450, 'guild_id': '779792159976456263', 'channel_id': '779806780686860298', 'message_flags': 0, 'message_id': '1044657613842948237', 'application_id': '574652751745777665', 'data': {'component_type': 2, 'custom_id': 'fish 351523222'}, 'session_id': 'hCVww8pfL78y0CyQTO6dOeDQFVWf6EpK'}
Headers: {'Date': 'Tue, 22 Nov 2022 16:56:34 GMT', 'Content-Type': 'application/json', 'Content-Length': '165', 'Connection': 'keep-alive', 'set-cookie': '__dcfduid=9ec7d3f46a8611eda74f42ea93164de7; Expires=Sun, 21-Nov-2027 16:56:34 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/, __sdcfduid=9ec7d3f46a8611eda74f42ea93164de76d7ba930435670c0e66881d28967ecfa59247ad1033e5c13f0e2e0933fc5902e; Expires=Sun, 21-Nov-2027 16:56:34 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/, __cfruid=a7cd672694dc9241056403af5ee98bed490ccaaf-1669136194; path=/; domain=.discord.com; HttpOnly; Secure; SameSite=None', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'x-envoy-upstream-service-time': '113', 'Via': '1.1 google', 'Alt-Svc': 'h3=":443"; ma=86400, h3-29=":443"; ma=86400', 'CF-Cache-Status': 'DYNAMIC', 'Report-To': '{"endpoints":[{"url":"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=yj6RnyVo1A6fJJdWZW4XOXtHkPXhoUX%2FLaq6eoQL%2BDTtnTxTpY27eJBacffANjITVjkf3OO9t1fMspHCukWOkH1M4waIFA%2B88KIAOenqgaVOhYNXYpOwCsB3EyBz"}],"group":"cf-nel","max_age":604800}', 'NEL': '{"success_fraction":0,"report_to":"cf-nel","max_age":604800}', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '76e332fbaad52784-PRG'} | [Stack] [FrameInfo(frame=<frame at 0x000001EB83A1A340, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\utils.py', line 50, code log>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\utils.py', lineno=50, function='log', code_context=["            log_data = f'\\n[{convert_time(time())}] {id} | [Traceback] {trace()} | [Event] {event} | [Stack] {stack()}\\n'\n"], index=0, positions=Positions(lineno=50, end_lineno=50, col_offset=110, end_col_offset=117)), FrameInfo(frame=<frame at 0x000001EB8394BCC0, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\api_wrapper.py', line 397, code request>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\api_wrapper.py', lineno=397, function='request', code_context=["                        debugger.log(err_message, 'ax0')\n"], index=0, positions=Positions(lineno=397, end_lineno=397, col_offset=24, end_col_offset=56)), FrameInfo(frame=<frame at 0x000001EB83A95300, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\autofishbot.py', line 288, code run>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\autofishbot.py', lineno=288, function='run', code_context=['                        if self.session.request(message_id=self.message.id, custom_id=self.message.play_id, category=BUTTON):\n'], index=0, positions=Positions(lineno=288, end_lineno=288, col_offset=27, end_col_offset=124)), FrameInfo(frame=<frame at 0x000001EB83AF2980, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 975, code run>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=975, function='run', code_context=['                self._target(*self._args, **self._kwargs)\n'], index=0, positions=Positions(lineno=975, end_lineno=975, col_offset=16, end_col_offset=57)), FrameInfo(frame=<frame at 0x000001EB83AF2380, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 1038, code _bootstrap_inner>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=1038, function='_bootstrap_inner', code_context=['                self.run()\n'], index=0, positions=Positions(lineno=1038, end_lineno=1038, col_offset=16, end_col_offset=26)), FrameInfo(frame=<frame at 0x000001EB83A89FE0, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 995, code _bootstrap>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=995, function='_bootstrap', code_context=['            self._bootstrap_inner()\n'], index=0, positions=Positions(lineno=995, end_lineno=995, col_offset=12, end_col_offset=35))]

[18:27 - 11/22/22] ax1 | [Traceback] [] | [Event] Code: 400

Response: {'message': 'Unknown Channel', 'code': 10003}
Request: {'type': 3, 'nonce': 1044665353407299645, 'guild_id': None, 'channel_id': '779806780686860298', 'message_flags': 0, 'message_id': '1044665636099461253', 'application_id': '574652751745777665', 'data': {'component_type': 2, 'custom_id': 'fish 1047928891'}, 'session_id': 'DT3gteDxtlQP4wlWZZftDLwE5nD1mFyU'}
Headers: {'Date': 'Tue, 22 Nov 2022 17:28:31 GMT', 'Content-Type': 'application/json', 'Content-Length': '45', 'Connection': 'keep-alive', 'set-cookie': '__dcfduid=15a4332e6a8b11ed9a2d1e1e9add1fd1; Expires=Sun, 21-Nov-2027 17:28:31 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/, __sdcfduid=15a4332e6a8b11ed9a2d1e1e9add1fd1ffa4cac485ca309b71e4e8b94d68fce1638159e0654090d39fd18d06f2fb0bd5; Expires=Sun, 21-Nov-2027 17:28:31 GMT; Max-Age=157680000; Secure; HttpOnly; Path=/, __cfruid=1c77f0a5512b3d3cf380c4bdee92dfad0d6dbeda-1669138111; path=/; domain=.discord.com; HttpOnly; Secure; SameSite=None', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'x-envoy-upstream-service-time': '42', 'Via': '1.1 google', 'Alt-Svc': 'h3=":443"; ma=86400, h3-29=":443"; ma=86400', 'CF-Cache-Status': 'DYNAMIC', 'Report-To': '{"endpoints":[{"url":"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=XeHEux9ryUO56TbYTXFW%2BIpNeLz7afHHsICDasqXq%2F2IzgpgHrCQgd5eOLe9iC3Ow3cI7pwkLFzsRlanmCx4xDolH1heBn85lvK0TsgtFQ4vgd7dIKCkaCyn%2FPfL"}],"group":"cf-nel","max_age":604800}', 'NEL': '{"success_fraction":0,"report_to":"cf-nel","max_age":604800}', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '76e361cbd8614131-PRG'} | [Stack] [FrameInfo(frame=<frame at 0x000001EB83A1A440, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\utils.py', line 50, code log>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\utils.py', lineno=50, function='log', code_context=["            log_data = f'\\n[{convert_time(time())}] {id} | [Traceback] {trace()} | [Event] {event} | [Stack] {stack()}\\n'\n"], index=0, positions=Positions(lineno=50, end_lineno=50, col_offset=110, end_col_offset=117)), FrameInfo(frame=<frame at 0x000001EB8394BCC0, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\api_wrapper.py', line 401, code request>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\app\\api_wrapper.py', lineno=401, function='request', code_context=["                        debugger.log(err_message, 'ax1')\n"], index=0, positions=Positions(lineno=401, end_lineno=401, col_offset=24, end_col_offset=56)), FrameInfo(frame=<frame at 0x000001EB83A95300, file 'C:\\Users\\niksa\\Desktop\\autofishbot-main\\autofishbot.py', line 288, code run>, filename='C:\\Users\\niksa\\Desktop\\autofishbot-main\\autofishbot.py', lineno=288, function='run', code_context=['                        if self.session.request(message_id=self.message.id, custom_id=self.message.play_id, category=BUTTON):\n'], index=0, positions=Positions(lineno=288, end_lineno=288, col_offset=27, end_col_offset=124)), FrameInfo(frame=<frame at 0x000001EB83AF2980, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 975, code run>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=975, function='run', code_context=['                self._target(*self._args, **self._kwargs)\n'], index=0, positions=Positions(lineno=975, end_lineno=975, col_offset=16, end_col_offset=57)), FrameInfo(frame=<frame at 0x000001EB83AF2380, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 1038, code _bootstrap_inner>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=1038, function='_bootstrap_inner', code_context=['                self.run()\n'], index=0, positions=Positions(lineno=1038, end_lineno=1038, col_offset=16, end_col_offset=26)), FrameInfo(frame=<frame at 0x000001EB83A89FE0, file 'C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', line 995, code _bootstrap>, filename='C:\\Users\\niksa\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\threading.py', lineno=995, function='_bootstrap', code_context=['            self._bootstrap_inner()\n'], index=0, positions=Positions(lineno=995, end_lineno=995, col_offset=12, end_col_offset=35))]

**Comment by DomeSs97** [comment]
There is link for full error.log. 

https://pastebin.com/p9zMTBcX


now got 1st. time this message from VF. 

maybe they have new antibot? 

Virtual Fisher
:information_source: Please only use your most recent Fish Again buttons as you may miss one of our antibot captchas otherwise.

**Comment by thejoabo** [comment]
Hey @DomeSs97 , thanks for sending the logs.

**Note: I edited your comment because it contains your Discord token. If that's not a disposable account, I strongly recommend that you change your password (in order to reset your token), otherwise anyone with access to it can do whatever they want to your account.**

> Virtual Fisher
> ℹ️ Please only use your most recent Fish Again buttons as you may miss one of our antibot captchas otherwise.

About this message, it's not a new captcha or anything like that, but in the attempt to reduce the amount of slash commands sent by the bot (they trigger api rate limits errors way faster than simple interactions) I removed a part of a function that resets the message id and play id when a message without the "Fish Again" embedded is sent by Virtual Fisher, therefore, the bot would keep the previous successful ids and press that button again (and just updating it when a new message with the "fish again" button appear). That's why you are seeing this message (this will be changed soon). 


> [18:27 - 11/22/22] ax1 | [Traceback] [] | [Event] Code: 400
> Response: {'message': 'Unknown Channel', 'code': 10003}

About these logs, the only important error is this "Unknown Channel" error, but I still have to debug it to understand how the guild_id could be "lost" mid-session (needs more debugging and testing)


> There is link for full error.log.
> https://pastebin.com/....

There is a lot of weird things here. 
- first of all, you might be missing a lot of captchas because the OCR.SPACE API servers are very unstable lately, their free API service is very slow for some reason ([status page](https://status.ocr.space/)) - that explain the absurd amount of "Captcha - request timeout" errors
- you are getting a concerning amount of "Receiver - run - reconnection" errors, I tried turning  my internet off and on while playing and was able to reproduce this error:

> {'code': 50035, 'errors': {'data': {'_errors': [{'code': 'COMPONENT_VALIDATION_FAILED', 'message': 'Component validation failed'}]}}, 'message': 'Invalid Form Body'}

I also got this message right after my connection is reestablished:

> ℹ️ Please only use your most recent Fish Again buttons as you may miss one of our antibot captchas otherwise.

- so, to summarize, your internet might be oscillating and causing the bot to loose track of messages (for missing some events)
- also, please avoiding sending manual commands via Discord while the bot is running, because it can cause exactly the behavior seen in the first error in your pastebin logs ("[01:29 - 11/22/22] Receiver - run (Message sent by the AFB while captcha is detected)")

Anyways, I saved your pastebin log to understand why the guild_id is being set to None and why the bot might be sending messages after a captcha is detected. I'll be releasing a patch when I have a solution to it (I'm quite busy for now).

**Comment by DomeSs97** [comment]
Ok. I will change pass and token. But this account is only for VF. On main Discord acc i have ban for 1year :D for using Version 1.x.x :D 


My connection is stable and without dropouts. I have pure 1gbps via fiber optic cable. 

VFB tomorrow I will upload a VPS where there is a stable connection with 20gbps.

If I can help with testing write to discord: DomeSs#8893

**Comment by thejoabo** [comment]
Quick fix 'ax1' at [Patch experimental-2.0.0.2](https://github.com/thejoabo/autofishbot/commit/ceb879bf5f023652e6050f3da0d2c91be22f1a93). I inspected the reconnection events and it was overwriting the guild id parameter. Reconnections might not be always caused by user-side network oscillations, other factors must be taken into account.
